flink-ml-lib/src/main/java/org/apache/flink/ml/classification/linear/LogisticRegressionModel.java

```diff
@@ -0,0 +1,185 @@
+        modelData = inputs[0];
+        return this;
+    }
+
+    @Override
+    public Table[] getModelData() {
+        return new Table[] {modelData};
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Table[] transform(Table... inputs) {
+        Preconditions.checkArgument(inputs.length == 1);
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) inputs[0]).getTableEnvironment();
+        DataStream<Row> inputStream = tEnv.toDataStream(inputs[0]);
+        final String broadcastModelKey = "broadcastModelKey";
+        DataStream<LogisticRegressionModelData> modelData =
+                LogisticRegressionModelData.getModelDataStream(this.modelData);
+        RowTypeInfo inputTypeInfo = TableUtils.getRowTypeInfo(inputs[0].getResolvedSchema());
+        RowTypeInfo outputTypeInfo =
+                new RowTypeInfo(
+                        ArrayUtils.addAll(
+                                inputTypeInfo.getFieldTypes(),
+                                BasicTypeInfo.DOUBLE_TYPE_INFO,
+                                TypeInformation.of(DenseVector.class)),
+                        ArrayUtils.addAll(
+                                inputTypeInfo.getFieldNames(),
+                                getPredictionCol(),
+                                getRawPredictionCol()));
+        DataStream<Row> predictionResult =
+                BroadcastUtils.withBroadcastStream(
+                        Collections.singletonList(inputStream),
+                        Collections.singletonMap(broadcastModelKey, modelData),
+                        inputList -> {
+                            DataStream inputData = inputList.get(0);
+                            return inputData.transform(
+                                    "doPrediction",
+                                    outputTypeInfo,
+                                    new PredictOperator(broadcastModelKey, getFeaturesCol()));
+                        });
+        return new Table[] {tEnv.fromDataStream(predictionResult)};
+    }
+
+    /** A utility operator used for prediction. */
+    private static class PredictOperator
+            extends AbstractUdfStreamOperator<Row, AbstractRichFunction>
+            implements OneInputStreamOperator<Row, Row> {
+
+        private final String broadcastModelKey;
+
+        private final String featuresCol;
+
+        private DenseVector coefficient;
+
+        public PredictOperator(String broadcastModelKey, String featuresCol) {
+            super(new AbstractRichFunction() {});
+            this.broadcastModelKey = broadcastModelKey;
+            this.featuresCol = featuresCol;
+        }
+
+        @Override
+        public void processElement(StreamRecord<Row> streamRecord) {
+            if (coefficient == null) {
+                LogisticRegressionModelData modelData =
+                        (LogisticRegressionModelData)
+                                userFunction
```

Comment:
nits: can we call `getRuntimeContext()` directly without using `userFunction`?

Reply:
For now we have to use `userFunction.getRuntimeContext()` because only the `RuntimeContext` of the `RichFunction` contains the broadcast variable. 

(It is due to the implementation of `BroadcastUtils#withBroadcast` variable, and also consistent with that in `DataSet` runtime).

Reply:
Hmm... is this our long term plan for how the algorithm developer should use `BroadcastUtils#withBroadcast`? What is the rational of doing so?

It seems counter-intuitive that `userFunction.getRuntimeContext() != getRuntimeContext()`. Note that this belongs to the public API (not the implementation) of `BroadcastUtils#withBroadcast` because this is what we expect algorithm developer to take care of. Is this behavior documented?

Reply:
Hmm, I don't think this is the long term plan to support `withBroadcast`. Now `withBroadcast` is marked as internal.

For the long run, we may need to discuss how to extend the existing implementations for `BroadcastState` and there may be a flip for this.

Reply:
The code has been updated to use `datastream.map(new PredictLabelFunction(), outputType)`, which is more clean to me. Please check out for `LogisticRegressionModel#L125`.

Also, can you update Knn similarly? @yunfengzhou-hub

Reply:
Sounds good. Let's discuss the long term solution for supporting `BroadcastUtils` later.